### DISCUSSION TITLE
RPI-RF-MOD stacked with Raspee2 problems

### DISCUSSION THREAD

**Discussion by beowulfof** [OpenCCU/OpenCCU#1812] [🇺🇸 Q&A] [answered]
I tried to converge my raspberrymatic server with my zigbee controller. Therefore I modded the RPI-RF-MOD with stackable headers and stacked the raspee2 deconz zigbee module on top.

Booting with only the RPI-RF-MOD works fine, but when the raspee2 is connected on top, only the hw-clock module of the raspbee2 is detected, RPI-RF-MOD is marked NA in this combination.

Any help appreciated to get this working, if possible.

![Bildschirmfoto von 2022-04-22 12-40-16](https://user-images.githubusercontent.com/20900257/164701168-4f2de560-a544-41d3-a00a-5e015a78324f.png)
![Bildschirmfoto von 2022-04-22 12-46-54](https://user-images.githubusercontent.com/20900257/164701173-5c197149-6bc6-43c3-b9a3-770b908621f1.png)
.

**Comment by jens-maus** [comment, marked as answer]
Sorry, but even without knowing the technical details of the zigbee HAT you are trying to use on a stackable GPIO header, this cannot work. Most probably the zigbee HAT is trying to use the same GPIO pins (e.g. the serial UART pins) like the RPI-RF-MOD and thus this cannot work.

Apart from that it is also not adviceable and also does not make sense to try to put all kind of RF HATs on the same interface (GPIO) because this only asks for trouble. And even if you would be able to get the RPI-RF-MOD working with the zigbee HAT you would probably run into other trouble like RF interferences between the RPI-RF-MOD and zigbee module that would negatively affect rf reception.

So better keep these rf modules separated or better use different interfaces (GPIO for X and USB foe Y) is the better approach here!

**Comment by beowulfof** [comment]
OK, I'd thought thats what the UART is for, since other realtime-clock (tiny-RTC) modules are stacked upon that as well and work just fine.

**Reply by jens-maus** [reply]
> OK, I'd thought thats what the UART is for, since other realtime-clock (tiny-RTC) modules are stacked upon that as well and work just fine.

Nope. RTC HATs don't use UART for communication but use the I2C with different addresses so that more than one device can be accessed. But a serial bus connection (UART) is something that only one application can occupy. Thus, this won't work!

**Reply by beowulfof** [reply]
My bad, you're absolutely right.
